src/icf/genCairo.ts

```diff
@@ -0,0 +1,220 @@
+    '\n\n//funtions to interact with given cairo contract\n' +
+    interactiveFuncs.join('\n')
+  );
+}
+
+function getStarknetLangDirective(): string[] {
+  return ['%lang starknet'];
+}
+
+function getForwarderInterface(abi: AbiType): string[] {
+  return [
+    '@contract_interface',
+    'namespace Forwarder {',
+    ...abi.map((item: AbiItemType) => {
+      if (item.type === 'function' && item.name !== '__default__') {
+        return [
+          `${INDENT}func ${item.name}(`,
+          ...item.inputs.map((input: { name: string; type: string }) => {
+            return `${INDENT}${INDENT}${input.name}: ${input.type},`;
+          }),
+          `${INDENT}) -> (${item.outputs
+            .map((output: { name: string; type: string }) => `${output.name}:${output.type}`)
+            .join(', ')}){`,
+          `${INDENT}}`,
+        ].join('\n');
+      }
+      return '';
+    }),
+    '}',
+  ];
+}
+
+export function getInteractiveFuncs(
+  abi: AbiType,
+  contract_address: string | undefined,
+  class_hash: string | undefined,
+): [string[], string[], StructAbiItemType[], string[], string[], StructAbiItemType[]] {
+  const structDependency = getStructDependencyGraph(abi);
+  const typeToStruct = typeToStructMapping(structDependency);
+  const [transformedStructs, transformedStructsFuncs, structTuplesMap] =
+    uint256TransformStructs(structDependency);
+
+  const imports: string[] = [
+    'from starkware.cairo.common.uint256 import Uint256',
+    'from starkware.cairo.common.cairo_builtins import HashBuiltin',
+    'from warplib.maths.utils import felt_to_uint256, narrow_safe',
+    'from warplib.maths.external_input_check_ints import warp_external_input_check_int256',
+  ];
+
+  const interactiveFuncs: string[] = [];
+  const expInpFunctionsMap: Map<string, string> = new Map();
+
+  abi.forEach((item: AbiItemType) => {
+    if (item.type === 'function' && item.name !== '__default__') {
+      const decorator: string = item.stateMutability === 'view' ? '@view' : '@external';
+
+      const callToFunc = (isDelegate: boolean) =>
+        `${INDENT}let (${item.outputs.reduce(
+          (acc: string[], output: { name: string; type: string }) => {
+            if (output.type.endsWith('*')) {
+              acc.pop();
+              acc.push(`${output.name}_len`);
+            }
+            return [...acc, `${output.name}_cast_rev`];
+          },
+          [],
+        )}) = Forwarder.${(isDelegate ? 'library_call_' : '') + item.name}(${
+          isDelegate ? class_hash ?? 0 : contract_address ?? 0
+        },${item.inputs.reduce((acc: string[], input: { name: string; type: string }) => {
+          if (input.type.endsWith('*')) {
+            acc.pop();
+            acc.push(`${input.name}_len`);
+          }
+          return [...acc, `${input.name}_cast`];
+        }, [])});`;
+
+      const funcDef = (isDelegate: boolean) => [
+        decorator,
+        `func _ITR_${
+          (isDelegate ? '_delegate_' : '') + item.name
```

Comment:
Why are generating two functions for each cairo function, one with the _delegate_?